### ISSUE TITLE
Feature Request: MSSQL message handler support

### ISSUE THREAD

**Issue by shgohh** [mkleehammer/pyodbc#558]
There are many ways output will be sent during execution in Microsoft SQL Server, such as:-
- Print statement will sent user-defined message to the output buffer
- Insert / Update statement (SET NOCOUNT OFF) will shows the count of the number of rows affected, to output buffer
- Select statement will return all selected rows, to output buffer

Note that Microsoft SQL Server buffers the output, this buffer must be flush to prevent buffer overflow. In my experience, subsequent code in the script will not be executed and no error will be thrown, when the buffer is overflowed.

pymssql provided this support through set_msghandler - message handler callback function, as follow:-

```
import _mssql

def my_msg_handler(msgstate, severity, srvname, procname, line, msgtext):
    """
    Our custom handler -- It simpy prints a string to stdout assembled from
    the pieces of information sent by the server.
    """
    print("my_msg_handler: msgstate = %d, severity = %d, procname = '%s', "
          "line = %d, msgtext = '%s'" % (msgstate, severity, procname,
                                         line, msgtext))

cnx = _mssql.connect(server='SQL01', user='user', password='password')
try:
    cnx.set_msghandler(my_msg_handler)  # Install our custom handler
    cnx.execute_non_query("USE mydatabase")  # It gets called at this point
finally:
    cnx.close()

```

Reference: http://pymssql.org/en/stable/_mssql_examples.html#custom-message-handlers

**Comment by gordthompson** [comment]
Related: #495

**Comment by gordthompson** [comment]
Fixed in #765
